### ISSUE TITLE
bug in life course report

### ISSUE THREAD

**Issue by f-hafner** [odissei-lifecourse/life-sequencing-dutch#54]
when running the life course report on the gron set, I get errors. working on branch `evaluate-full-dataset`.

```bash
python write_embedding_metadata.py --collection_name gron_embedding_set
python generate_life_course_report.py --collection_name gron_embedding_set --savename "results/myfile_" 
```

#### With `report_parts = [5.1]` 
I get an error on the cross-validation in line 714 of `linear_variable_prediction`, which is raised from `sklearn.model_selection._validation`:

```bash
ValueError:
All the 5 fits failed.
It is very likely that your model is misconfigured.
You can try to debug the error by setting error_score = "raise" 
```

#### With `report_parts = [3]` I get:
```bash
File "*/.../report_utils.py", line 753, in linear_variable_prediction
ValueError: not enough values to unpack (expected 3, got 0)
```

Since we had the report for gron running for the network embeddings, I would first try to debug this one. 

For completeness, I also created an LLM embedding set for groningen only. then, with `report_parts=[5.1]`,  I get:

```bash
ValueError:
All the 5 fits failed.
It is very likely that your model is misconfigured.
You can try to debug the error by setting error_score = "raise" 

Below are more details about the failures:
-----------------------------------------------------------------
5 fits failed with the following error:
File "../sklear/linear_model/_logistic.py", line 1246, in fit
ValueError: this solver needs samples of at least 2 classees in the data, but the data contains only one class: 1
```

**Comment by f-hafner** [comment]
Looking into the behavior of gron network embeddings for section 5.1
- cross validation fails for year 2023 when `baseline` is not `None`. the reason is that there is only one level in the outcome variables, `1`. cross-validation works for year 2023 when `baseline` is `None`.
- I don't understand how we even get to year 2023; in line 105 of `generate_life_course_report.py` the years are specified from 2011 to 2021

**Comment by f-hafner** [comment]
Update
- it is unclear where these errors have come from. they could be from subsetting the data to Gron, or from using the work environment
- the initial bug not reported here was that embedding lengths of person and partner differed. the reason was a  list addition with `x+=y` which updated also the object to which `x` was pointing, which was not intended. The bug must have been around for a while

**Comment by f-hafner** [comment]
Update II 
Running evaluation on the embeddings of model medium  v0.0.2, running `linear_variable_prediction` for income (I think section 3 of the report):

```python
File "/.../report_utils.py", line 735, in linear_variable_prediction
scores = cross_val_score(model, full_embedding_list, full_label_list, cv5)
[...]
File "/../sklearn/model_selection/_validation.py", ValueError: all the 5 fits failed. It is very likely that your model is misconfigured.

Below are more details about the failures:
-------------------------------------------------------------
[...]
File "/../sklarn/utils/validation.py", line 1049, in check_array
_assert_all_finite(
[...]
ValueError: Input contains infinity or a value too large for dtype("float16").

```

This now looks more like an NaN problem in the embeddings. I will look into this at the next allocation on Monday.

**Comment by f-hafner** [comment]
> This now looks more like an NaN problem in the embeddings. I will look into this at the next allocation on Monday.

This was caused by the dtype of the embeddings, which is now float16 from the new pretraining specification. I fixed it and it works for medium model v0.0.2.

**Comment by f-hafner** [comment]
Update III

In part 3, when baseline is not None, I stil get 

```bash
File "*/.../report_utils.py", line 753, in linear_variable_prediction
ValueError: not enough values to unpack (expected 3, got 0)
```

Inspecting the baselines and embeddings, I see that except for 2011, the lists are empty. I think this is related to #21 , and currently the code for evaluating on the income prediction is broken. Once I confirm that the code runs on the other parts, I will close this issue and make a separate one for section 3.
